File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/testing/specs/GivenAFaultingConnection/WhenCheckingIfTheConnectionIsPossible.java
package com.lasthopesoftware.bluewater.client.connection.testing.specs.GivenAFaultingConnection;

import com.lasthopesoftware.bluewater.client.connection.specs.FakeConnectionProvider;
import com.lasthopesoftware.bluewater.client.connection.testing.ConnectionTester;
import com.lasthopesoftware.bluewater.shared.promises.extensions.specs.FuturePromise;
import org.junit.BeforeClass;
import org.junit.Test;

import java.io.IOException;
import java.util.concurrent.ExecutionException;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

public class WhenCheckingIfTheConnectionIsPossible {

	private static boolean result;

	@BeforeClass
	public static void before() throws ExecutionException, InterruptedException {
		final ConnectionTester connectionTester = new ConnectionTester();
		final FakeConnectionProvider connectionProvider = new FakeConnectionProvider();
		connectionProvider.mapResponse(p -> { throw new IOException(); }, "Alive");
		result = new FuturePromise<>(connectionTester.promiseIsConnectionPossible(connectionProvider)).get();
	}

	@Test
	public void thenTheResultIsCorrect() {
		assertThat(result).isFalse();
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/connection/testing/GivenAFaultingConnection/WhenCheckingIfTheConnectionIsPossible.kt
package com.lasthopesoftware.bluewater.client.connection.testing.GivenAFaultingConnection

import com.lasthopesoftware.bluewater.client.connection.FakeConnectionProvider
import com.lasthopesoftware.bluewater.client.connection.testing.ConnectionTester
import com.lasthopesoftware.bluewater.shared.promises.extensions.toExpiringFuture
import org.assertj.core.api.AssertionsForClassTypes.assertThat
import org.junit.jupiter.api.Test
import java.io.IOException

class WhenCheckingIfTheConnectionIsPossible {

	private val result by lazy {
		val connectionTester = ConnectionTester
		val connectionProvider = FakeConnectionProvider()
		connectionProvider.mapResponse({ throw IOException() }, "Alive")
		connectionTester.promiseIsConnectionPossible(connectionProvider).toExpiringFuture().get()!!
	}

	@Test
	fun `then the result is correct`() {
		assertThat(result).isFalse
	}
}
